### PR TITLE
Update to reflect 2024 milestones

### PR DESCRIPTION
**Pull request by jhaaaa**
Preview: https://junk-range-possible-git-update-roadmap-2024-xmtp-labs.vercel.app/roadmap

### CODE REVIEW

File: src/pages/roadmap.md [hunk 1]
```diff
@@ -2,46 +2,50 @@
...
-### 2023 Q3
+### Q1
 
-- Onchain transaction reference content type
-- React SDK in Production
-- React Native SDK in Production (currently in Beta)
-- Android SDK in Production (currently in Beta)
-- XMTP v3 in Alpha
+- Group chat beta available in iOS, Android, and React Native SDKs
+- Initial support for push notifications, message history backups/portable inboxes, and smart contract wallet accounts
```
Comment (jhaaaa) [inline src/pages/roadmap.md:23]: In some cases, we list items at alpha, beta, or production. For some items we don't - should we always list the release status? For example, in the case of this bullet -- does "initial" mean alpha?

File: src/pages/roadmap.md [hunk 2]
```diff
@@ -2,46 +2,50 @@
...
+- Contact directory available in production
+- Invite inbox and conversation consent smart contract prototypes available in alpha
 
-### 2023 Q4
+### Q2 
 
-- XMTP v3 in Beta
+- Group chat available in JavaScript and React SDKs
+- Message history synchronization for new devices
```
Comment (jhaaaa) [inline src/pages/roadmap.md:30]: How is this item related to "Initial support for ... message history backups/portable inboxes"?

File: src/pages/roadmap.md [hunk 3]
```diff
@@ -2,46 +2,50 @@
...
+- Key revocation and rejection of messages from revoked identities
+- Enable 1:1 conversations to use MLS and `libxmtp`
+- Flutter SDK support
```
Comment (jhaaaa) [inline src/pages/roadmap.md:33]: Do we want to specify support for what?

File: src/pages/roadmap.md [hunk 4]
```diff
@@ -2,46 +2,50 @@
...
+- Testnet blockchain supporting contacts, invites, inboxes, and conversations
```
Comment (jhaaaa) [inline src/pages/roadmap.md:34]: What is the difference between "testnet blockchain" and "production-ready testnet" below? Is the testnet blockchain different from the testnet?

File: src/pages/roadmap.md [hunk 5]
```diff
@@ -2,46 +2,50 @@
...
+## 2023
 
-We typically follow this release cycle:
+### Q4
 
-- Alpha: Super early peek into the feature being built. Not ready for use in production apps.
-- Beta: Opportunity to test, validate, and provide feedback. Not ready for use in production apps.
-- Production: Ready for use in production apps.
+- Group chat prototype powered by Messaging Layer Security (MLS)
+- Shared allow/deny lists for iOS/Android SDKs
```
Comment (nplasterer) [inline src/pages/roadmap.md:14]: I think we are going to deliver this in all SDKS

File: src/pages/roadmap.md (hunk 1 above)
Comment (nplasterer) [reply src/pages/roadmap.md:23]: We already to some degree support push notifications should there be a distinction here? Like push notifications backed by MLS? Or maybe thats not important.

File: src/pages/roadmap.md [hunk 6]
```diff
@@ -2,46 +2,50 @@
...
+### Q1
 
-- Onchain transaction reference content type
-- React SDK in Production
-- React Native SDK in Production (currently in Beta)
-- Android SDK in Production (currently in Beta)
-- XMTP v3 in Alpha
+- Group chat beta available in iOS, Android, and React Native SDKs
+- Initial support for push notifications, message history backups/portable inboxes, and smart contract wallet accounts
+- Contact directory available in production
```
Comment (nplasterer) [inline src/pages/roadmap.md:24]: Does this mean the contact directory is available on mainnet for ethereum? Or does production mean it's piped all the way through the protocol and being used in the SDK. @jac18281828

File: src/pages/roadmap.md (hunk 3 above)
Comment (nplasterer) [reply src/pages/roadmap.md:33]: This I believe is when flutter was slotted to get group chat. But I believe we should probably bump this to Q3 to be safe. cc @bwcDvorak for flutter work visibility.

File: src/pages/roadmap.md (hunk 4 above)
Comment (nplasterer) [reply src/pages/roadmap.md:34]: Per my comment above https://github.com/xmtp/xmtp-dot-org/pull/615/files#r1411398237 I don't understand how we can be using the contact register in production without it being on mainnet.

File: src/pages/roadmap.md (hunk 1 above)
Comment (saulmc) [reply src/pages/roadmap.md:23]:
```suggestion
- Initial MLS support for push notifications and message history backups/portable inboxes
- Support for smart contract wallet accounts
```

Push notifs/portable history are scoped to MLS, since they are already supported in v2. SC accounts are new across the board.

**Comment by jhaaaa** [comment]
Starting with a refresh take based on public epics
